_posts/2025-06-24-kafka-queue.md

```diff
@@ -0,0 +1,703 @@
+          "deliveryState": 4,
+          "deliveryCount": 5
+        }
+      ]
+    }
+  }
+}
+```
+
+After 4 attempts of delivery, the record was still in <b>Available</b> (enum 0) state but after the 5th attempt, it went into <b>Archived</b> (enum 4) state, which is why it was not retried again. 
+
+### Broker and share configurations
+
+<i>Summary of the configurations mentioned in this blog:</i>
+
+| Configuration | Type | Default | Description |
+| :--: |:-------------| :-------------| :-------------|
+| group.coordinator.rebalance.protocols  | Broker   | classic,consumer | It should be set to `classic,consumer,share` to enable share group. |
+| unstable.api.versions.enable           | Broker   | false | It should be set to true, in order to use this feature until it is production ready .|
+| share.coordinator.state.topic.replication.factor | Broker | 3 | Replication factor for the share-group state topic. Setting this greater than the cluster size will result in failure to create the topic and share group consumers will not work. |
+| group.share.record.lock.duration.ms    | Broker   | 30s | The record acquisition lock duration in milliseconds for share groups. |
+| group.share.partition.max.record.locks | Broker   | 200 | Share group record lock limit per share-partition. |
+| group.share.delivery.count.limit       | Broker   | 5   | The maximum number of delivery attempts for a record delivered to a share group. |
+| group.share.session.timeout.ms         | Broker   | 45000 | The timeout to detect client failures when using the share group protocol. |
+| fetch.max.bytes                        | Broker   | 57671680 | The maximum number of bytes returned for a fetch request. |
+| max.poll.interval.ms                   | Consumer | 300000 | The maximum delay between invocations of `poll()` when using consumer group management. |
+| max.poll.records                       | Consumer | 500 | The maximum number of records returned in a single `poll()`. | 
+| share.acknowledgement.mode             | Consumer | implicit | Controls the acknowledgement mode for a share consumer. If set to implicit, must not use `acknowledge()`. Instead, delivery is acknowledged in the next `poll()` call. If set to explicit, must use  `acknowledge()` to acknowledge delivery of records. 
+
+The broker configurations can be set in your Kafka CR like shown in the example above. And the consumer configurations can be set in your client application, like shown in the Java API example.
+
+### The current limitations of the feature
+
+This feature is currently not recommended for production clusters as of Apache Kafka 4.0.0 but can be tried out with a test cluster as demonstrated above. If you are trying out this feature on your existing development/test cluster, note that you will not be able to upgrade to Apache Kafka 4.1 or later. You would have to disable this feature first and delete the state topic, in order to upgrade to a later version.
```

Comment:
> note that you will not be able to upgrade to Apache Kafka 4.1 or later

@ppatierno Curious how does this show up in your 4.1.0-RC PR. Did you test it? 😯

Reply:
No I didn't test it. I didn't know about this feature "blocking" the upgrade. @tinaselenge is it somewhere in official Kafka doc?